### DISCUSSION TITLE
building a narrative

### DISCUSSION THREAD

**Discussion by anarcat** [OpenVoxProject/planning#11] [Ideas]
So I see people starting to say that [we're forking Puppet](https://fosstodon.org/@genebean/113664896831791581).

# Who's forking what?

I'd like us to take a different tack: we're not forking Puppet; **Perforce is forking Puppet**.

Puppet is, [according to Wikipedia][] "an open source configuration management utility by Puppet, Inc". What Perforce is doing right now is taking the open source code that we have collaborative used, debugged, written, collaborated, stared at and deployed on thousands of machines, and closing access to it to paying customers.

[according to Wikipedia]: https://en.wikipedia.org/wiki/Puppet_(disambiguation)

They will stop producing publicly available binary packages, and will stop publicly distributed the source for that software. They claim to not change the license of the software, but that's really an insignificant detail because, effectively, the source code for Puppet, as produced by Perforce, will no longer be publicly available.

What's worse, they have explicitly stated that they will not allow us to use the name Puppet, even though they explicitly want us to manage the "community" and "open source" versions of Puppet, which, when you think about it, makes absolutely no sense.

So, again: **we are not forking Puppet, Perforce is forking Puppet, and they don't want us to use the name.** Which is why we're having this silly conversation about renaming Puppet (#9).

# About future collaboration

Perforce has repeatedly said they want to collaborate and create a democratic process, but I've now sat in three of those town hall meetings about this, and they have never taken the community's input into account. Every one of those meetings has been effectively a set of slides and PR material trying to get us to work on their opensource version without giving us any tools (like owning the GitHub organisation or trademarks) to do so.

I think everyone here deeply cares (and needs!) Puppet. I am here because I manage a fleet of 100 machines for the Tor Project. The Debian Project, Riseup, Koumbit, and so many other organisations are in the same boat. We can't just get off this Puppet ship. So we're going to keep doing what we were doing before Perforce took over, which is to collaborate over code and issues, and make this product better.

I also suspect everyone here is open to *true* collaboration with Perforce. If there's anyone there that still cares about the ecosystem, I suspect we will have people happy to sit on committees to discuss roadmaps and language design. But at this point, what's actually happening, is we're told what's going to happen, we're not being asked, we're not being included in the process, so I doubt that Perforce is *actually* interested in the collaboration they claim to be doing.

It's a little late to talk about collaboration and democratic process when [Voxpupuli](https://voxpupuli.org/) have established such [excellent stewardship mechanisms for Puppet modules](https://voxpupuli.org/elections/) (which, I suspect, will yield similar processes here) and, in parallel, Perforce has utterly failed to even *propose* such a process for how the community should be managed.

We are, again, doing the work that needs to be done for this community, nothing more. We're doing this in a transparent, open, and collaborative way (see https://github.com/OpenPuppetProject/planning/discussions/10), and everyone is welcome to join us, including Perforce.

This is not a hostile fork, this is keeping Puppet alive.

**Comment by ekohl** [comment]
On collaboration they mentioned it would be a program similar to the [trusted contributor program](https://www.puppet.com/ecosystem/contribute/trusted-contributors) and as someone who is in that program to contribute to the modules I can say with confidence that it's not collaboration. Often Perforce employees just rush in and change things or take decisions without any transparency. I had hoped to raise this during the townhall, but we ran out of time. On the other hand, I doubt it would have mattered. Tzvika clearly showed he wasn't listening, just repeating his own talking points over and over again.
